src/audio/pcm_converter/pcm_converter_generic.c

```diff
@@ -161,6 +161,101 @@ static int32_t _pcm_convert_i_to_f(int32_t src)
 
 #endif /* CONFIG_FORMAT_FLOAT && (CONFIG_FORMAT_S16LE || CONFIG_FORMAT_S24LE || CONFIG_FORMAT_S32LE) */
 
+#if CONFIG_FORMAT_FLOAT && CONFIG_FORMAT_S16LE
+static void pcm_convert_s16_to_f(struct comp_buffer *source,
+				struct comp_buffer *sink, uint32_t samples)
+{
+	int16_t *src;
+	int32_t *dst; /* float */
```

Comment:
is there a reason why we can't use float * ?

I can predict someone will be listening to a buffer of integers that's full of floating point content.
I'd recommend the comp_buffer structure has a type qualifier so that we can avoid errors when dealing with int32 or floats...

Reply:
@plbossart Added @singalsu to the review as well. We are bypassing the compiler here and doing the float conversions on our own. I do wonder a bit how well this works if one wants to port float algorithms, and how portable this is form hw-arch to another (this is now in hw-agnostic part of sof codebase).